elastic/kibana
# [Alerting] Improve creation and editing of "Elasticsearch query" rule in Management

Closes https://github.com/elastic/kibana/issues/134183

## Summary

This PR brings an improved and more convenient UI for creating rules to Management page as we have it now on Discover page.

![Jun-27-2022 15-43-11](https://user-images.githubusercontent.com/1415710/175956213-3bd7ed5b-8c8d-4c50-9dca-7010ef0b1680.gif)


Steps to test:
- From Stack Management page:
  - Open "Rules and Connectors" 
  - Select "Elasticsearch query" from rules list
  - Select how to configure this new rule: KQL/Lucene or Query DSL
  - Check that both options work
- From Discover page:
  - Press "Alerts" > "Create search threshold rule" 
  - Check that it still allows to configure a rule in KQL/Lucene mode
  
Stack Management > New rule > Elasticsearch Query
<img width="500" alt="Screenshot 2022-06-22 at 16 09 21" src="https://user-images.githubusercontent.com/1415710/175050636-3b64d5d8-29c1-49a8-a0eb-c51b6649c069.png">

"KQL or Lucene" is selected
<img width="500" alt="Screenshot 2022-06-22 at 16 09 29" src="https://user-images.githubusercontent.com/1415710/175050776-26bb973f-0292-409c-b950-6bc4b7eddc78.png">

"Query DSL" is selected
<img width="500" alt="Screenshot 2022-06-22 at 16 09 45" src="https://user-images.githubusercontent.com/1415710/175050862-25a64142-c4c6-472f-a14a-bf42d5436f5c.png">

Discover > Alerts > Create search threshold rule
<img width="500" alt="Screenshot 2022-06-22 at 16 08 07" src="https://user-images.githubusercontent.com/1415710/175050995-655ff801-2960-4f78-9b3f-94e954f5086e.png">



### Checklist
